apollo-router/src/services/subgraph_service.rs

```diff
@@ -110,151 +139,257 @@ impl tower::Service<crate::SubgraphRequest> for SubgraphService {
 
-            let path = schema_uri.path().to_string();
-            let response = client
-                .call(request)
-                .instrument(tracing::info_span!("subgraph_request",
-                    "otel.kind" = "CLIENT",
-                    "net.peer.name" = &display(host),
-                    "net.peer.port" = &display(port),
-                    "http.route" = &display(path),
-                    "http.url" = &display(schema_uri),
-                    "net.transport" = "ip_tcp",
-                    "apollo.subgraph.name" = %service_name
-                ))
-                .await
-                .map_err(|err| {
-                    tracing::error!(fetch_error = format!("{:?}", err).as_str());
+        Box::pin(make_calls)
+    }
+}
 
-                    FetchError::SubrequestHttpError {
-                        service: service_name.clone(),
-                        reason: err.to_string(),
-                    }
-                })?;
-            // Keep our parts, we'll need them later
-            let (parts, body) = response.into_parts();
-            if display_headers {
-                tracing::info!(
-                    http.response.headers = ?parts.headers, apollo.subgraph.name = %service_name, "Response headers from subgraph {service_name:?}"
-                );
-            }
-            if let Some(content_type) = parts.headers.get(header::CONTENT_TYPE) {
-                if let Ok(content_type_str) = content_type.to_str() {
-                    // Using .contains because sometimes we could have charset included (example: "application/json; charset=utf-8")
-                    if !content_type_str.contains(APPLICATION_JSON.essence_str())
-                        && !content_type_str.contains(GRAPHQL_JSON_RESPONSE_HEADER_VALUE)
-                    {
-                        return if !parts.status.is_success() {
-                            Err(BoxError::from(FetchError::SubrequestHttpError {
-                                service: service_name.clone(),
-                                reason: format!(
-                                    "{}: {}",
-                                    parts.status.as_str(),
-                                    parts.status.canonical_reason().unwrap_or("Unknown")
-                                ),
-                            }))
-                        } else {
-                            Err(BoxError::from(FetchError::SubrequestHttpError {
-                                service: service_name.clone(),
-                                reason: format!("subgraph didn't return JSON (expected content-type: {} or content-type: {GRAPHQL_JSON_RESPONSE_HEADER_VALUE}; found content-type: {content_type:?})", APPLICATION_JSON.essence_str()),
-                            }))
-                        };
-                    }
-                }
+/// call_http makes http calls with modified graphql::Request (body)
+/// with or without the query and hash acc. to APQ need.
```

Comment:
```suggestion
```
(this part of the code does not care about APQ, it's only for the HTTP call and response deserialization)